dali/pipeline/executor/executor2/exec_node_task.cc

```diff
@@ -259,6 +259,7 @@ void OpTask::RunOp() {
     ResetInputLayouts();
     PropagateSourceInfo(*ws_);
   }
+  if (!ws_->GetIterationData()) throw std::runtime_exception("Failed to acquire Iteration Data.");
```

Comment:
It's not "acquired". It can't possibly be null. If it was, it would not be a run-time error but a coding error.
```suggestion
  assert(ws_->GetIterationData());
```